source/contracts/reporting/Auction.sol

```diff
@@ -0,0 +1,239 @@
+        currentAttoEthBalance = initialAttoEthBalance;
+
+        currentUpperBoundRepPriceInAttoEth = 0;
+        currentLowerBoundRepPriceInAttoEth = 0;
+
+        currentAuctionIndex = _currentAuctionIndex;
+
+        initialRepSalePrice = lastRepPrice.mul(Reporting.getAuctionInitialPriceMultiplier());
+        initialEthSalePrice = Reporting.getAuctionInitialPriceMultiplier().mul(10**36).div(lastRepPrice);
+        return true;
+    }
+
+    function tradeRepForEth(uint256 _attoEthAmount) public initializeNewAuctionIfNeeded returns (bool) {
+        require(!bootstrapMode);
+        require(currentAttoEthBalance > 0);
+        require(_attoEthAmount > 0);
+        _attoEthAmount = _attoEthAmount.min(currentAttoEthBalance);
+        uint256 _ethPriceInAttoRep = getEthSalePriceInAttoRep();
+        uint256 _attoRepCost = _attoEthAmount.mul(_ethPriceInAttoRep) / 10**18;
+        reputationToken.trustedAuctionTransfer(msg.sender, this, _attoRepCost);
+        msg.sender.transfer(_attoEthAmount);
+        uint256 _attoEthSold = initialAttoEthBalance.sub(currentAttoEthBalance);
+        uint256 _newTotalAttoEthSold = _attoEthSold.add(_attoEthAmount);
+        uint256 _repPriceInAttoEth = (10**36) / _ethPriceInAttoRep;
+        currentLowerBoundRepPriceInAttoEth = currentLowerBoundRepPriceInAttoEth
+            .mul(_attoEthSold)
+            .add(_attoEthAmount.mul(_repPriceInAttoEth))
+            .div(_newTotalAttoEthSold);
+        currentAttoEthBalance = currentAttoEthBalance.sub(_attoEthAmount);
+        currentRepPrice = currentLowerBoundRepPriceInAttoEth.add(currentUpperBoundRepPriceInAttoEth) / 2;
+        return true;
+    }
+
+    function tradeEthForRep(uint256 _attoRepAmount) public initializeNewAuctionIfNeeded payable returns (bool) {
+        require(currentAttoRepBalance > 0);
+        require(_attoRepAmount > 0);
+        _attoRepAmount = _attoRepAmount.min(currentAttoRepBalance);
+        uint256 _repPriceInAttoEth = getRepSalePriceInAttoEth();
+        uint256 _attoEthCost = _attoRepAmount.mul(_repPriceInAttoEth) / 10**18;
+        // This will raise an exception if insufficient ETH was sent
+        msg.sender.transfer(msg.value.sub(_attoEthCost));
+        reputationToken.transfer(msg.sender, _attoRepAmount);
+        uint256 _attoRepSold = initialAttoRepBalance.sub(currentAttoRepBalance);
+        uint256 _newTotalAttoRepSold = _attoRepSold.add(_attoRepAmount);
+        currentUpperBoundRepPriceInAttoEth = currentUpperBoundRepPriceInAttoEth
+            .mul(_attoRepSold)
+            .add(_attoRepAmount.mul(_repPriceInAttoEth))
+            .div(_newTotalAttoRepSold);
+        currentAttoRepBalance = currentAttoRepBalance.sub(_attoRepAmount);
+        // We don't update the price while in bootstrap mode since its a one sided auction
+        if (!bootstrapMode) {
+            currentRepPrice = currentLowerBoundRepPriceInAttoEth.add(currentUpperBoundRepPriceInAttoEth) / 2;
+        }
```

Comment:
How much code could we simplify if we skipped bootstrap mode and just seeded the system with some ETH?  (lets ignore for a moment where we get that ETH)

Reply:
It wasn't too complicated but that would be nicer of course.

Reply:
One option would be for FF to just fund the initial auction.  Probably too expensive though.  Another option would be to have a separate mechanism for turning initial REP into ETH rather than trying to bundle it into the existing auction system.

Reply:
That seems more complicated to me to create a whole other system. There's really not much additional code to do the bootstrap